sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -188,6 +191,10 @@ def __init__(
             HttpLoggingPolicy(**kwargs),
         ]
 
+        if self.aad_credentials:
+            scopes = base.create_scope_from_url(self.url_connection)
+            policies.append(CosmosBearerTokenCredentialPolicy(self.aad_credentials, scopes))
```

Comment:
I'm don't think we usually append the credentials policy to the end of the pipeline - could be worth checking where azure-core typically puts the authentication policy (e.g. before or after the retry policy)

Reply:
pics from tables and storage sdks, I moved it to be before the DecodePolicy like in theirs:
<img width="255" alt="image" src="https://user-images.githubusercontent.com/30335873/160183750-cd705494-f9a9-442a-80b5-7d42261a6807.png">
<img width="286" alt="image" src="https://user-images.githubusercontent.com/30335873/160183895-25b7653a-ba59-47a3-9c56-60819bd9b005.png">

Reply:
I would start with using azure-core as your reference for things like this, which actually seems to have it _after_ the retry policy:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/_pipeline_client.py#L131
You could test it both to see what impact retry might have on resetting the auth header.

Reply:
Kay, let's leave it where it is for now - we can do a little homework before GA to figure out why it sits in different places in different SDKs to see if it will make a difference here.